fuel_test/deploy_env.sh

```diff
@@ -134,6 +136,9 @@ function wait_for_node {
 			return
 		fi
 		let counter=counter+1
+		if [ $(($counter % 10)) -eq 0 ]; then
+			ssh -qo StrictHostKeyChecking=no root@$xs_host 'xe vm-reboot vm="'$vm'" force=true'
```

Comment:
Since retry_interval is passed in to this fucntion, the reboot command will happen at an undefined time.  If you want to reboot, you should have some additional check (e.g. a time-elapsed setting) or wrap wait_for_node in a wait_for_node_reboot_and_retry method, which would invoke wait_for_node several times, rebooting inbetween each attempt.